kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/VersionInfo.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.proxy;
+
+import java.io.IOException;
+import java.util.Objects;
+import java.util.Properties;
+
+import org.slf4j.LoggerFactory;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+
+public interface VersionInfo {
+    VersionInfo VERSION_INFO = getVersionInfo();
+
+    String version();
+
+    String commitId();
+
+    String commitMessage();
+
+    private static VersionInfo getVersionInfo() {
+
+        try (var resource = Info.class.getClassLoader().getResourceAsStream("META-INF/metadata.properties")) {
+            if (resource != null) {
+                Properties properties = new Properties();
+                properties.load(resource);
+                String version = properties.getProperty("kroxylicious.version", Info.UNKNOWN);
+                String commitId = properties.getProperty("git.commit.id", Info.UNKNOWN);
+                String commitMessage = properties.getProperty("git.commit.message.short", Info.UNKNOWN);
+                return new Info(version, commitId, commitMessage);
+            }
+        }
+        catch (IOException e) {
+            var logger = LoggerFactory.getLogger(Info.class);
+            logger.warn("Failed to retrieve version information (ignored)", e);
+        }
+        return Info.UNKNOWN_VERSION_INFO;
+    }
+
+    final class Info implements VersionInfo {
```

Comment:
Is this not duplicating a similar class in the proxy property?

Reply:
I didn't spot it - which one exactly?